shared/domains/chelonia/internals.js

```diff
@@ -2091,6 +2151,19 @@ const handleEvent = {
     }
     this.config.reactiveSet(state.contracts[contractID], 'HEAD', hash)
     this.config.reactiveSet(state.contracts[contractID], 'height', height)
+    const missingDecryptionKeyIdsForMessage = missingDecryptionKeyIdsMap.get(message)
+    if (Array.isArray(missingDecryptionKeyIdsForMessage)) {
+      let missingDecryptionKeyIds = state.contracts[contractID].missingDecryptionKeyIds
+      if (!missingDecryptionKeyIds) {
+        missingDecryptionKeyIds = []
+        this.config.reactiveSet(state.contracts[contractID], 'missingDecryptionKeyIds', missingDecryptionKeyIds)
+      }
+      missingDecryptionKeyIdsForMessage.forEach(keyId => {
+        if (missingDecryptionKeyIds.includes(keyId)) return
+        missingDecryptionKeyIds.push(keyId)
+      })
+      this.config.reactiveSet(state.contracts[contractID], 'height', height)
+    }
```

Comment:
Could you please add a comment above this block here explaining:

1. What this is doing and why
2. Why it's being done here